lara-typescript/src/interactive-api-lara-host/iframe-saver.ts

```diff
@@ -206,6 +230,53 @@ export class IFrameSaver {
     });
   }
 
+  public saveMetadata(metadata: object) {
+    if (!this.learnerStateSavingEnabled()) { return; }
+
+    if (JSON.stringify(metadata) === JSON.stringify(this.metadata)) {
```

Comment:
`JSON.stringify` is not idempotent, i.e. it's not guaranteed to stringify two different objects with the same contents identically. In particular, the order of properties is not guaranteed. If you mean for this to handle the case of two different equivalent objects (i.e. when `metadata !== this.metadata`), then you may want to use something like [deep-equal](https://www.npmjs.com/package/deep-equal) instead.

Reply:
True. I think I'll just remove this check, as it doesn't seem to be worth it anyway. There are no use cases when this optimization would be used in practice.